### PR TITLE
Fix #446: expose StreamConstaintsException as ApiException

### LINKED ISSUE #446: Create dataset based fallout configurations
Create dataset based fallout configurations.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Improves exception handling for specific StreamConstraintsException-caused failures, to make them appear as regular `JsonApiException`.

Also fixes a few other existing cases where we expose `JsonMappingException` instead of `JsonApiException`, to reduce exposure of irrelevant/unnecessary internal details.

**Which issue(s) this PR fixes**:
Fixes #466

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/exception/WebApplicationExceptionMapper.java [hunk 1]
```diff
@@ -12,8 +15,23 @@
 public class WebApplicationExceptionMapper {
 
   @ServerExceptionMapper
-  public RestResponse<CommandResult> genericExceptionMapper(WebApplicationException e) {
-    Throwable toReport = null != e.getCause() ? e.getCause() : e;
+  public RestResponse<CommandResult> webApplicationExceptionMapper(WebApplicationException e) {
+    // 06-Nov-2023, tatu: Let's dig the innermost root cause; needed f.ex for [jsonapi#448]
+    //    to get to StreamConstraintsException
+    Throwable toReport = e;
+    while (toReport.getCause() != null) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/exception/WebApplicationExceptionMapper.java:22]: Used to only dig into immediate root cause, but some cases have at least 2 levels of nesting (StreamConstraintsException -> JsonMappingException -> JsonApiException f.ex).

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/CountIntegrationTest.java [hunk 2]
```diff
@@ -336,7 +337,10 @@ public void withExistFalseOperator() {
           .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
           .then()
           .statusCode(200)
-          .body("errors[1].message", is("$exists operator supports only true"));
+          .body("errors", hasSize(1))
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/CountIntegrationTest.java:340]: Was exposed not as `JsonApiException`, test wasn't validating type.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/DeleteOneIntegrationTest.java [hunk 3]
```diff
@@ -148,12 +148,10 @@ public void noOptionsAllowed() {
           .body("data.document", is(nullValue()))
           .body("status", is(nullValue()))
           .body("errors", is(notNullValue()))
-          .body("errors", hasSize(2))
+          .body("errors", hasSize(1))
           .body("errors[0].message", is("Command accepts no options: DeleteOneCommand"))
-          .body("errors[0].exceptionClass", is("JsonMappingException"))
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/DeleteOneIntegrationTest.java:153]: Due to interaction or various catch blocks, had `JsonApiException` "hidden" inside `JsonMappingException` (wrong way around). But we also do not necessarily want to expose internal/intermediate `JsonMappingException` at all.

**Review by Yuqi-Du — APPROVED** [review]
LGTM

**Pull request merged**